cmd/controller/main.go

```diff
@@ -43,6 +48,20 @@ func main() {
 	lo.Must0(op.AddHealthzCheck("cloud-provider", awsCloudProvider.LivenessProbe))
 	cloudProvider := metrics.Decorate(awsCloudProvider)
 
+	cm := &v1.ConfigMap{}
+	if err := op.GetClient().Get(ctx, types.NamespacedName{Name: "karpenter-global-settings"}, cm); err != nil {
+		logging.FromContext(ctx).Error(err)
+	}
+
+	k8sVersion, err := op.AMIProvider.KubeServerVersion(ctx)
+	if err != nil {
+		logging.FromContext(ctx).Error(err)
+	}
+	err = kompat.IsCompatible("hack/compatibility-karpenter.yaml", cm.Labels["app.kubernetes.io/version"], k8sVersion)
```

Comment:
I don't think we should be inferring the k8s version like this. We should be getting the K8s version from the API server similar to how it's done when we discover AMIs.